numpy/ma/core.py

```diff
@@ -786,22 +786,17 @@ def fix_invalid(a, mask=nomask, copy=True, fill_value=None):
 
 class _DomainCheckInterval(object):
     """
-    Define a valid interval, so that :
-
-    ``domain_check_interval(a,b)(x) == True`` where
-    ``x < a`` or ``x > b``.
+    Define a valid interval, so that:
 
+    ``_DomainCheckInterval(a,b)(x)`` is False where ``a <= x <= b``.
```

Comment:
I think the way forward might be to update the implementation to match these new comments - the only reason it's done the way it is is to avoid doing an not operation, which seems to come at the cost of `nan`-correctness